### PR TITLE
[🐛 BUG] Negating run properties with CLI arguments not working

### LINKED ISSUE #2663: [🐛 BUG] Negating run properties with CLI arguments not working
### 🔍 What went wrong?

Tried `taipy run --no-run-browser main.py` but it still opened in browser.

Same thing with `Gui.run(use_reloader=True)` and then `taipy run --no-reloader main.py`, the reloader is still used.

Seemingly only affects negation -- `taipy run --use-reloader main.py` works as expected.

### ✅ Expected Behavior

`--no-run-browser` and `--no-reloader` should disable the browser and reloader respectively.

### 🔄 Steps to Reproduce

1.  Create file main.py

```python
from taipy.gui import Gui
Gui(page="# CLI argument issue").run()
```

2. Run `taipy run --no-run-browser main.py`

Browser still opens.

### 📦 Taipy Version

develop

### 📋 Additional Context (Optional)

```bash

```

### 📜 Code of Conduct

- [x] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+) to avoid duplicates.
- [ ] I am willing to work on this issue (optional)

### ✅ Acceptance Criteria

- [ ] A reproducible unit test is added.
- [ ] Code coverage is **at least 90%**.
- [ ] The bug reporter validated the fix.
- [ ] Relevant **documentation updates** or an issue created in

### PR DESCRIPTION
**Pull request by KachanovYev**
## What type of PR is this? (Check all that apply)
- [ ] 🛠 Refactor
- [x] 🐛 Bug Fix
- [ ] ✨ Feature
- [ ] ⚡ Optimization
- [ ] 📝 Documentation Update

---

## Description

This PR addresses a CLI argument negation bug in Taipy. When running:

```bash
taipy run --no-run-browser main.py
```

the application still opens the browser, contrary to the expected behavior. Similarly, using:

```bash
taipy run --no-reloader main.py
```

does not prevent the reloader from being used. However, using positive flags (e.g., `--use-reloader`) works as intended.

This PR ensures that negated CLI flags (`--no-run-browser` and `--no-reloader`) correctly disable the browser launch and the reloader, respectively. The argument parsing logic has been refactored to properly handle negation for these flags.

---

## Related Tickets & Documents

- [[🐛 BUG] CLI negation flags not working as expected](https://github.com/Avaiga/taipy/issues/2663)

---

## How to Reproduce the Issue

1. Create a file named `main.py`:
    ```python
    from taipy.gui import Gui
    Gui(page="# CLI argument issue").run()
    ```
2. Run:
    ```bash
    taipy run --no-run-browser main.py
    ```
3. Observe: The browser still opens.

---

## Backporting

This change should be backported to:

- [ ] 3.0
- [ ] 3.1
- [ ] 4.0
- [x] develop

---

## Checklist

We encourage keeping the code coverage percentage at 80% or above.

- [x] This solution meets the acceptance criteria of the related issue.
- [x] The related issue checklist is completed.
- [x] This PR includes unit tests for the developed code.
  - If not, explain why: N/A
- [x] End-to-End tests have been added or updated.
  - If not, explain why: Manual testing in affected environments.
- [x] The documentation has been updated, or a dedicated issue has been created.
  - If not, explain why: Inline code comments provided.
- [x] The release notes have been updated.
  - If not, explain why: Will be added after review.

---

## Additional Context

- Bug only affects negation (`--no-run-browser`, `--no-reloader`)
- Positive flags work as expected (`--use-reloader`)
- See [Code of Conduct](#) for contribution guidelines.

---

## Acceptance Criteria

- A reproducible unit test is added.
- Code coverage is at least 90%.
- The bug reporter validated the fix.
- Relevant documentation updates or an issue created.

### CODE REVIEW

**Comment by KachanovYev** [comment]
@FredLL-Avaiga 
@FabienLelaquais 

Hi
I resolved issue. 
Please check when you have a time.

File: taipy/common/_cli/_run_cli.py [hunk 1]
```diff
@@ -54,7 +54,17 @@ def handle_command(cls):
             external_args.extend(all_args[external_args_index + 1 :])
             all_args = all_args[:external_args_index]
 
-        taipy_args = [f"--taipy-{arg[2:]}" if arg.startswith("--") else arg for arg in all_args]
+        def add_taipy_prefix(key: str):
+            """
+                Adds the '--taipy-' prefix to command-line arguments starting with '--'.
+                If the argument starts with '--no-', inserts 'taipy-' after '--no-'.
+                Otherwise, inserts 'taipy-' after '--'.
+                """
```
Comment (trgiangdo) [inline taipy/common/_cli/_run_cli.py]: Fix the indentation

```suggestion
            """Adds the '--taipy-' prefix to command-line arguments starting with '--'.
            If the argument starts with '--no-', inserts 'taipy-' after '--no-'.
            Otherwise, inserts 'taipy-' after '--'.
            """
```

File: taipy/gui/servers/flask/server.py [hunk 2]
```diff
@@ -377,7 +377,8 @@ def run(
             if client_url is not None:
                 client_url = client_url.format(port=port)
                 _TaipyLogger._get_logger().info(f" * Application is accessible at {client_url}")
-        if not self.is_running_from_reloader() and self._gui._get_config("run_browser", False):  # type: ignore[attr-defined]
+        if not self.is_running_from_reloader() and self._gui._get_config("run_browser",
+                                                                         False) is True:  # type: ignore[attr-defined]
```
[the remark this answers is not in the document]
Comment (trgiangdo) [reply taipy/gui/servers/flask/server.py]: I don't think this change is necessary.
```python
if self._gui._get_config("run_browser", False) is True:
```
and 
```python
if self._gui._get_config("run_browser", False):
```
is the same

**Review by trgiangdo — COMMENTED** [review]
Please take a look at the `tests/gui/config/test_cli.py` file.

There, we are having tests to test the effect of the CLI arguments  where we mock the CLI arguments using:
```python
with patch("sys.argv", ["prog", "--host", "my_4th_host", "--port", "5004"]):
```

It's true that we are missing tests for the negative args. Please add a new test for the `--no-run-browser` and `--no-reloader` arguments.

**Comment by KachanovYev** [comment]
Hi 
in need support with [issue of pipline](https://github.com/Avaiga/taipy/actions/runs/16592979579/job/46934334937?pr=2702)

File: taipy/gui/servers/flask/server.py [hunk 3]
```diff
@@ -377,7 +377,8 @@ def run(
             if client_url is not None:
                 client_url = client_url.format(port=port)
                 _TaipyLogger._get_logger().info(f" * Application is accessible at {client_url}")
-        if not self.is_running_from_reloader() and self._gui._get_config("run_browser", False):  # type: ignore[attr-defined]
+        if not self.is_running_from_reloader() and self._gui._get_config("run_browser",
+                                                                         False):  # type: ignore[attr-defined]
```
Comment (trgiangdo) [inline taipy/gui/servers/flask/server.py]: Please revert this change

**Comment by FredLL-Avaiga** [comment]
> Hi in need support with [issue of pipline](https://github.com/Avaiga/taipy/actions/runs/16592979579/job/46934334937?pr=2702)

I think it was not a real error => a re-run fixed it :-)

**Comment by KachanovYev** [comment]
> > Hi in need support with [issue of pipline](https://github.com/Avaiga/taipy/actions/runs/16592979579/job/46934334937?pr=2702)
> 
> I think it was not a real error => a re-run fixed it :-)

yes! 
Thank you.

**Review by trgiangdo — DISMISSED** [review]
There's 1 redundant change. Otherwise, LGTM.

File: taipy/gui/servers/flask/server.py (hunk 3 above)
Comment (KachanovYev) [reply taipy/gui/servers/flask/server.py]: resolved
pushed the code

**Review by trgiangdo — APPROVED** [review]
Awesome. @KachanovYev thank you for the contribution
